proposals/lip-0052.md

```diff
@@ -362,712 +379,1832 @@ The module provides the following commands to modify the NFT store.
+        },
+        "collectionID": {
+            "dataType": "bytes",
+            "length": COLLECTION_ID_LENGTH,
+            "fieldNumber": 3
+        },
+        "result": {
+            "dataType": "uint32"
+            "fieldNumber": 4
+        }
+    }
+}
+```
+
+#### destroy
+
+This event has `name = EVENT_NAME_DESTROY`, and is emitted when the [destroy](#destroy-2) function is called.
+
+##### Topics
+
+- `address`: the address of the NFT owner.
+- `nftID`: ID of the destroyed NFT.
+
+##### Data
+
+```java
+destroyEventDataSchema = {
+    "type": "object",
+    "required": [
+        "address",
+        "nftID",
+        "result"
+    ],
+    "properties": {
+        "address": {
+            "dataType": "bytes",
+            "length": ADDRESS_LENGTH,
+            "fieldNumber": 1
+        },
+        "nftID": {
+            "dataType": "bytes",
+            "length": NFT_ID_LENGTH,
+            "fieldNumber": 2
+        },
+        "result": {
+            "dataType": "uint32",
+            "fieldNumber": 3
+        }
+    }
+}
+```
+
+#### lock
+
+This event has `name = EVENT_NAME_LOCK`, and is emitted when the [lock](#lock-2) function is called.
+
+##### Topics
+
+- `module`: name of the module that locked the NFT.
+- `nftID`: ID of the locked NFT.
+
+##### Data
+
+```java
+lockEventDataSchema = {
+    "type": "object",
+    "required": [
+        "module",
+        "nftID",
+        "result"
+    ],
+    "properties": {
+        "module": {
+            "dataType": "string",
+            "minLength": MIN_MODULE_NAME_LENGTH,
+            "maxLength": MAX_MODULE_NAME_LENGTH,
+            "fieldNumber": 1
+        },
+        "nftID": {
+            "dataType": "bytes",
+            "length": NFT_ID_LENGTH,
+            "fieldNumber": 2
+        },
+        "result": {
+            "dataType": "uint32",
+            "fieldNumber": 3
+        }
+    }
+}
+```
+
+#### unlock
+
+This event has `name = EVENT_NAME_UNLOCK`, and is emitted when the [unlock](#unlock-2) function is called.
+
+##### Topics
+
+- `module`: name of the module that unlocked the NFT.
+- `nftID`: ID of the unlocked NFT.
+
+##### Data
+
+```java
+unlockEventDataSchema = {
+    "type": "object",
+    "required": [
+        "module",
+        "nftID",
+        "result"
+    ],
+    "properties": {
+        "module": {
+            "dataType": "string",
+            "minLength": MIN_MODULE_NAME_LENGTH,
+            "maxLength": MAX_MODULE_NAME_LENGTH,
+            "fieldNumber": 1
+        },
+        "nftID": {
+            "dataType": "bytes",
+            "length": NFT_ID_LENGTH,
+            "fieldNumber": 2
+        },
+        "result": {
+            "dataType": "uint32",
+            "fieldNumber": 3
+        }
+    }
+}
+```
+
+#### setAttributes
+
+This event has `name = EVENT_NAME_SET_ATTRIBUTES`, and is emitted when the [setAttributes](#setattributes-2) function is called.
+
+##### Topics
+
+- `module`: name of the module that sets the attributes.
```

Comment:
I don't think this is a useful topic and I would remove it. I don't see why someone should query all events related to a module.

Reply:
Agree. I don't yet have a perfect understanding of events, so thanks for this comment because it helps me understand the logic behind events.